Backend/app/Http/Middleware/CheckAdmin.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+
+class CheckAdmin
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  Request  $request
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if (Auth::check() && Auth::user()->is_admin) {
+            return $next($request);
+        }
+
+        return response()->json(["error" => "NO"], 403);
+    }
+}
```

Comment:
You don't have import for the Auth here. Visual studio code highlights to me the lack of all imports in this code